staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/validator.go

```diff
@@ -96,9 +99,23 @@ func (a customResourceValidator) ValidateUpdate(ctx context.Context, obj, old ru
 	return allErrs
 }
 
-// WarningsOnUpdate returns warnings for the given update.
-func (customResourceValidator) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
-	return nil
+var standardFinalizers = sets.NewString(
+	metav1.FinalizerKubernetes,
+	metav1.FinalizerOrphanDependents,
+	metav1.FinalizerDeleteDependents,
+)
+
+func validateKubeFinalizerName(stringValue string, fldPath *field.Path) field.ErrorList {
+	allWarnings := field.ErrorList{}
+	for _, msg := range apimachineryvalidation.IsQualifiedName(stringValue) {
+		allWarnings = append(allWarnings, field.Invalid(fldPath, stringValue, msg))
+	}
+	if len(strings.Split(stringValue, "/")) == 1 {
+		if !standardFinalizers.Has(stringValue) {
+			allWarnings = append(allWarnings, field.Invalid(fldPath, stringValue, "name is neither a standard finalizer name nor is it fully qualified"))
```

Comment:
I'd drop the language about "standard finalizer names" and explain *why* we want finalizer names to be domain qualified:

```suggestion
			allWarnings = append(allWarnings, field.Invalid(fldPath, stringValue, "prefer a domain-qualified finalizer name to avoid accidental conflicts with other finalizer writers"))
```